jsonlines/jsonlines.py

```diff
@@ -295,27 +305,17 @@ def write_all(self, iterable):
             self.write(obj)
 
 
-try:
-    from typing import Literal, overload
-
-    @overload
-    def open(name, **kwargs) -> Reader:
-        ...
+@overload
+def open(file: _Openable, mode: Literal["r"], **kwargs) -> Reader:
```

Comment:
If we change to `mode: Literal["r"] = "r"` then the type annotations will appear when the arg isn't specified.

Reply:
thanks, addressed in 05b810f67eaeae25a48089a204aa8b46711e820e